server/middleware/barcode/barcodeAuthorisationMiddleware.ts

```diff
@@ -0,0 +1,32 @@
+import { RequestHandler } from 'express'
+import type { VerifyErrors, JwtPayload } from 'jsonwebtoken'
+import { verify } from 'jsonwebtoken'
+import config from '../../config'
+import logger from '../../../logger'
+
+export default function barcodeAuthorisationMiddleware(): RequestHandler {
+  return (req, res, next) => {
+    res.locals.barcodeUser = undefined
+
+    if (!req.session.createBarcodeAuthToken) {
+      logger.warn('Attempt to access protected barcode resource without an auth token')
+      return res.redirect('/link/request-link')
+    }
+
+    const token = req.session.createBarcodeAuthToken
+    return verify(
+      token,
+      config.barcodeTokenPublicKey,
+      { algorithms: ['RS256'] },
+      (err: VerifyErrors, payload: JwtPayload) => {
+        if (err) {
+          logger.warn('Unable to verify auth token', token, err.message)
```

Comment:
If the error is that the token has expired then this is BAU - so not sure we need to warn. For any other errors - such as signed incorrectly - this is very interesting but might also be nice to see the payload to give us a clue as to where the bogus token came from?

Reply:
I've removed the log statement for now (as per other logging comments) . As you say, we can re-address logging and metrics in a subsequent ticket 👍